### PR TITLE
Upgrade to Opentelemetry v0.5.0

### PR DESCRIPTION
**Pull request by cmouli84**

### CODE REVIEW

**Comment by cmouli84** [comment]
Created this PR for issue #12

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,6 @@ go 1.13
 
 require (
 	github.com/newrelic/newrelic-telemetry-sdk-go v0.2.0
-	go.opentelemetry.io/otel v0.4.3
+	go.opentelemetry.io/otel v0.5.0
```
Comment (MrAlias) [inline go.mod:7]: Any chance we can bump this to `v0.6.0`? That is the latest from OTel and I think the changes you include here are all that are needed for that version as well.

Comment (MrAlias) [reply go.mod:7]: Ah, I forgot about the switch to new metric instruments in `v0.6.0`. That switch should be tackled in another PR.

Comment (cmouli84) [reply go.mod:7]: Sounds good.

**Comment by cmouli84** [comment]
@MrAlias I requested review by mistake. Please approve and merge the Pull request if the changes look fine

**Comment by MrAlias** [comment]
@cmouli84 I'm in approval of this, but don't have permissions needed to merge. That would need someone from the @newrelic/go-agent team. Feel free to request a review from them.

**Comment by cmouli84** [comment]
@purple4reina Can you please review this change to upgrade to opentelemetry v0.5.0

**Pull request merged**